atlasdb-commons/src/main/java/com/palantir/async/initializer/AsyncInitializing.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.async.initializer;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.processors.DoDelegate;
+
+public interface AsyncInitializing {
+    @DoDelegate
+    default boolean isInitialized() {
+        return isInitializedAsync().isDone() && !isInitializedAsync().isCancelled();
```

Comment:
This gets a future twice in the case where it's done, should we extract it to a variable?

Reply:
Done